drivers/SmartThings/harman-luxury/src/init.lua

```diff
@@ -120,8 +120,9 @@ end
 local function check_for_updates(device)
   log.trace(string.format("%s, checking if device values changed", device.device_network_id))
   local ip = device:get_field(const.IP)
-  local changes, _ = api.InvokeGetUpdates(ip)
-  if changes then
+  local changes, err = api.InvokeGetUpdates(ip)
+  -- check if changes is empty
+  if not err and type(changes) == "table" and next(changes) ~= nil then
```

Comment:
Both the `type(changes) == "table"` and `next(changes) ~= nil` checks are duplicated in the body of this if statement, so they are useless. They should be removed from here or the other place.

Reply:
You are absolutely right. Thanks!